### ISSUE TITLE
Introduction to Position Token - Y3d Mining Pool V2 Future Contract Design Document

### ISSUE THREAD

**Issue by lychees** [carboclan/y3d-protocol#15]
## TL;DR:
In order to make the underlying market more liquid, we'd like to make the liquidity token even less liquid.
People who hold their y3d position may want to quit even if suffer from `5%` withdraw fee at some moment.
But if he/she can find some one in the market who want to join, he/she may transfer his/her position to the new holder with some discount

By tokenized the y3dlized position, the scenario above has the potential to turn out to be true.

## Technical detail
The changes are minor. We upgrade the solc version from `0.5.x` to `0.6.x`, then import the `ERC20.sol` standard template in line 21.

The source code can be found here: [https://github.com/carboclan/y3d-protocol/tree/master/contracts/pools/v2](https://github.com/carboclan/y3d-protocol/tree/master/contracts/pools/v2).

```javascript=21
import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
```
And use the `_mint()` and `_burn()` in the `stake()` and `withdraw()` function. So that people could transfer their position token to someone else or even build a uniswap trade pair for better liquidity and price discovery.

```javascript=168
    function stake(uint256 amount) update(msg.sender) virtual public {
        _mint(msg.sender, amount);
        LPT.safeTransferFrom(msg.sender, address(this), amount);
    }

    function withdraw(uint256 amount) update(msg.sender) virtual public {
        _burn(msg.sender, amount);
        uint256 tax = amount.div(20); if (exodus == true) tax = 0;
        amount = amount.sub(tax);
        if (amount > LPT.balanceOf(address(this))) ICrvDeposit(crv_deposit).withdraw(amount - LPT.balanceOf(address(this)));
        LPT.safeTransfer(msg.sender, amount);
        make_profit(tax);
    }
```

**Comment by lychees** [comment]
Deprecated.
